### ISSUE TITLE
Retrieve most recent entries

### ISSUE THREAD

**Issue by ldgrp** [Guthers/snails#9]
The current API spec lacks the ability to retrieve the most recent entries (questions). Changes to the *API specification*, *database*, and the *server* will be necessary to make this happen.

The proposed endpoint is as follows

`GET /api/entries`

Parameters
```
count : Optional[int]  # number of entries to retrieve. Default value is 10
before_id: Optional[EntryId]  # returned entries must be older than this entry
after_id: Optional[EntryId] # returned entries must be more recent than this entry 
```

Response
`Entry[]  # a list of entries`

**Comment by ldgrp** [comment]
The second parameter **must** be an entry id. Suppose two entries are submitted at the same time with entry ids 20 and 21, respectively. 

Assume an **incorrect** implementation that takes a datetime for the `from` parameter.
The client requests the 20 most recent posts (1-20). It would check the last post and ask for 10 more posts older than `entry 20`. The server would skip `entry 21` and return entries 21-31.

**Comment by ldgrp** [comment]
It is also useful to retrieve the entries _more recent_ than a given entry. 

The issue above has been edited to reflect this.

**Comment by ldgrp** [comment]
Here is an excellent resource on cursor based pagination

https://slack.engineering/evolving-api-pagination-at-slack/

**Comment by Joshua-Hwang** [comment]
I'll get on this when we start implementing entries. Doesn't look like it should cause much issues on the database or server ends.

The easiest solution is to retrieve all entries then *slowly* filter which entries we want.

We could even extend this to our other entry-like features: messaging and news.

**Comment by ldgrp** [comment]
Implemented in #16
